Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,167 @@
+            this.get('container').setHTML(this.template());
+            this._renderBlocks();
+
+            return this;
+        },
+
+        /**
+         * Toggles blocks active state
+         *
+         * @method _toggleBlocksActiveState
+         * @protected
+         * @param event {Object} event facade
+         */
+        _toggleBlocksActiveState: function (event) {
+            this.get('blocks').forEach(function (block) {
+                block.set('active', event.newVal);
+            });
+        },
+
+        /**
+         * Renders blocks on the dashboard
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this.get('blocks').forEach(function (block) {
+                fragment.append(block.render().get('container'));
+            });
+
+            this.get('container').one(SELECTOR_CONTENT).setHTML(fragment);
+        },
+
+        /**
+         * Sorts the provided list of blocks by priority.
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Array} list of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            return blocks.sort(function (a, b) {
+                return a.get('priority') > b.get('priority');
+            });
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param view {Y.eZ.DashboardBlockBaseView} dashboard block view instance
+         * @return {Array} the list of blocks
+         */
+        addBlock: function (view) {
+            var blocks = this.get('blocks'),
+                identifier = view.get('identifier');
+
+            if (!identifier || !(view instanceof Y.eZ.DashboardBlockBaseView)) {
+                return blocks;
+            }
+
+            if (this._findBlock(identifier)) {
+                this.removeBlock(identifier);
+            }
+
+            view.addTarget(this);
+            view.set('active', this.get('active'));
+
+            blocks.push(view);
+
+            blocks = this._sortBlocks(blocks);
+
+            this._set('blocks', blocks);
+
+            return blocks;
+        },
+
+        /**
+         * Finds block with a given identifier
+         *
+         * @method _findBlocks
+         * @param identifier {String} block identifier
+         * @return {Y.eZ.DashboardBlockBaseView} a block view
+         */
+        _findBlock: function (identifier) {
+            return this.get('blocks').filter(function (block) {
+                return block.get('identifier') === identifier;
+            })[0];
+        },
+
+        /**
+         * Removes a block from the dashboard
+         *
+         * @method removeBlock
+         * @param identifier {String} block identifier
+         * @return {Array} the list of blocks
```

Comment:
same as https://github.com/ezsystems/PlatformUIBundle/pull/584#discussion_r62698596 and https://github.com/ezsystems/PlatformUIBundle/pull/584/files#r62849728 don't return anything.